### PR TITLE
Add "guide" documents

### PR DESCRIPTION
**Pull request by pmuens**

### CODE REVIEW

File: docs/guide/creating-a-service.md [hunk 1]
```diff
@@ -1,14 +1,45 @@
...
-`cd` into a directory of your choice and run `serverless create --name my-service --provider aws`.
-Serverless will create a skeleton for your new Serverless service inside the `my-service` directory.
+Let's create our first Serverless service!
+
+## Creating our service
+
+`cd` into a directory of your choice and run `serverless create --name my-service --provider aws` to create a service
+skeleton with the service name `my-service` on your local machine.
+
+Services are provider specific. The `--provider aws` tells Serverless to create an Amazon Web Services dependent service.
```
Comment (mthenw) [inline docs/guide/creating-a-service.md]: I think it's worth to stay with "AWS" through the whole guide

File: docs/guide/creating-a-service.md [hunk 2]
```diff
@@ -1,14 +1,45 @@
...
+### [`serverless.env.yaml`](../understanding-serverless/serverless-env-yaml.md)
+
+This file includes all the environment variables Serverless uses. It automatically creates environment variables for a
+default stage called `dev` and a default region called `us-east-1`. Those variables will be referenced and used as a
+default later on for deployment.
+
+### `handler.js`
+
+The `handler.js` file includes a function skeleton which returns a simple message. The function definition in
+`serverless.yaml` will point to this handler.js file and the function inside of it.
```
Comment (mthenw) [inline docs/guide/creating-a-service.md]: `handler.js`

File: docs/guide/creating-a-service.md [hunk 3]
```diff
@@ -1,14 +1,45 @@
...
+
+## Conclusion
 
-Open the directory with your favorite editor. You should see some files. One of those is the `serverless.yaml` file.
-This file holds all the important information about your service. You should see e.g. a `functions` definition where
-one function is defined.
+We've just create our very first service with one simple `create` command. With this in place we're ready to deploy
```
Comment (mthenw) [inline docs/guide/creating-a-service.md]: created

Comment (mthenw) [inline docs/guide/creating-a-service.md]: not sure but probably "With that"

File: docs/guide/creating-a-service.md [hunk 4]
```diff
@@ -1,14 +1,45 @@
...
+
+You'll see the following files:
+- `serverless.yaml`
+- `serverless.env.yaml`
+- `handler.js`
+
+### [`serverless.yaml`](../understanding-serverless/serverless-yaml.md)
+
+This is our core service file. You can see the name of our service, the provider and the first function inside the
+`functions` definition which points to the `handler.js` file.
```
Comment (flomotlik) [inline docs/guide/creating-a-service.md]: There should also be a link here that goes the the specific definition (same as in the header) but specifically says that it goes to it, e.g. "If you want to read more about the details of your serverless.yml check out our [in-depth guide](../understanding-serverless/serverless-yaml.md)."

File: docs/guide/creating-a-service.md [hunk 5]
```diff
@@ -1,14 +1,45 @@
...
+
+### [`serverless.env.yaml`](../understanding-serverless/serverless-env-yaml.md)
+
+This file includes all the environment variables Serverless uses. It automatically creates environment variables for a
+default stage called `dev` and a default region called `us-east-1`. Those variables will be referenced and used as a
+default later on for deployment.
```
Comment (flomotlik) [inline docs/guide/creating-a-service.md]: Same as above, should have a call to the more in depth guide mentioned in the text so people know how to get there.

File: docs/guide/deploying-a-service.md [hunk 6]
```diff
@@ -1,4 +1,31 @@
...
-resources are deployed to AWS.
+Let's deploy our service on the corresponding cloud provider infrastructure we've defined for our service.
+
+## Deploying our service
+
+Make sure that you're still in the service directory.
+
+Run `serverless deploy` to start the deployment process.
+
+Serverless will now deploy the whole service (with all it's functions and events (we'll add some events soon)) to the
```
Comment (mthenw) [inline docs/guide/deploying-a-service.md]: ))

Comment (flomotlik) [inline docs/guide/deploying-a-service.md]: NEST ALL THE THINGS 😃

**Pull request merged**